src/component/list.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react';
+import { Form, ToDo } from './form';
+import { Item } from './item';
+
+export function List() {
+  const [todos, setToDos] = useState<ToDo[]>([]);
+  const [filter, setFilter] = useState<FilterOptions>(FilterOptions.All);
+
+  const onSubmit = (todo: ToDo) => {
+    setToDos([todo, ...todos]);
+  };
+
+  const onClick = () => {
+    setToDos([...todos]);
+  };
+
+  const show = (todos: ToDo[]) => {
+    const list = todos.map((todo) => {
+      return Item({ todo, filter, handleDeleteToDo})
+    });
+    return <ul onClick={onClick}>{list}</ul>;
+  };
+
+  const left = () => {
+    return todos.filter((todo) => !todo.isComplete).length;
+  };
+
+  const filtered = () => {
+    let array: ToDo[] = [];
+    switch (filter) {
+      case FilterOptions.All: {
+        array = todos;
```

Comment:
you can use ‘return todos.filter(...’ without assigning it to a variable and breaks and its also immutable

Reply:
Good catch, thanks!